### ISSUE TITLE
Performance optimisations of the geometry validation module (GmlGeoX)

### ISSUE THREAD

**Issue by jonherrmann** [etf-validator/governance#57]
**ETF Improvement Proposal (EIP)**

# Background and Motivation:

Large parts of the geometry validation module [GmlGeoX](https://github.com/etf-validator/etf-gmlgeox) have been rewritten to improve the validation performance, add new capabilities (e.g., for arcs) and improve the architecture / the maintainability of the code.

# Proposed change

See https://github.com/interactive-instruments/etf-gmlgeox/blob/EIPs/EIP-57/Readme.md for a detailed specification of the new API.

Since the API of the module changes significantly and will not be compatible with existing ETSs, the proposal is to release the new version as a major release (version 2). As a result, both the old and new version of GmlGeoX could be provided with the ETF software (as long as both versions are compatible with the BaseX version used in an ETF release).

The version 1 module would still be usable when the namespace of GmlGeoX version 1 is used:
```xquery 
import module namespace ggeo='de.interactive_instruments.etf.bsxm.GmlGeoX'
```

The new module would get a new namespace:
```xquery
import module namespace geox = 'https://modules.etf-validator.net/gmlgeox/2'
```

The library will be part of the BaseX testdriver, it will be integrated into the [etf-bsxtd](https://github.com/etf-validator/etf-bsxtd) repository.

# Alternatives
Support for version 1 could be removed. 

# Funding
Yes

# Additional information
n/a

**Comment by MarcoMinghini** [comment]
**SG meeting 24 (2020-08-06)**
It is agreed to accept the EIP and move it under _TC: scheduled for release_ in the [EIP project board](https://github.com/orgs/etf-validator/projects/2). The EIP will be included in the next release of the ETF.

The old version (v. 1) of the GmlGeoX module needs to be integrated with the new version of deegree, and this should be done in time for the next release. The approach for this integration should be discussed in a [new EIP](https://github.com/etf-validator/governance/issues/96); once the approach is proven to work for v. 1 of GmlGeoX, in the future the same approach can be extended to v. 2.

**Comment by jonherrmann** [comment]
Implemented in [Version 2.1.0](https://github.com/etf-validator/etf-webapp/releases/tag/2.1.0)
